addons/website_event/tests/test_event_internals.py

```diff
@@ -224,3 +226,87 @@ def test_registration_answer_search(self):
         # should fetch "registration_2" because the answer to the first question is "Q1-Answer2"
         # should fetch "registration_3" because the answer to the third question is "Answer2" (as free text)
         self.assertEqual(search_res, registration_2 | registration_3)
+
+    @users('portal')
+    def test_portal_user_is_visible_on_website(self):
+        """
+        Tests the computed field `is_visible_on_website` on `event.event`.
+        `filtered_domain` is used to check if the `_compute_is_visible_on_website` method is correct
+        `search` is used to check if the `_search_is_visible_on_website` method is correct
+        """
+
+        event_public, event_link, event_portal = [
+            self.env['event.event'].sudo().create(
+                {'name': 'event', 'website_visibility': website_visibility}
+            ) for website_visibility in ['public', 'link', 'logged_users']
+        ]
+        events_rset = event_public | event_link | event_portal
```

Comment:
Seems like this is copy/pasted in all three methods, maybe we could use a setUpClass to create "events_altered_visibility" instead?

You could also call them "event_public", "event_link_only" and "event_logged_users" and assert separately, would make it easier to read/follow.